src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/armor/RadiationTask.java

```diff
@@ -74,4 +87,39 @@ protected void onPlayerTick(Player p, PlayerProfile profile) {
             }
         }
     }
+
+    /**
+     * Checks if the {@link Player} is within their grace period. A grace period is granted after death
+     * to give enough time to remove the radioactive items before being killed once more, which
+     * with KeepInventory on would result in a very hard-to-escape loop.
+     * @see RadioactivityListener
+     *
+     * @param player
+     *              The {@link Player} to check against.
+     *
+     * @return Returns true if the {@link Player} is within their grace period.
+     */
+    private boolean withinGracePeriod(@Nonnull Player player) {
+        long gracePeriodEnd = ACTIVE_GRACE_PERIODS.getOrDefault(player.getUniqueId(), 0L);
+        if (gracePeriodEnd >= System.currentTimeMillis()) {
+            // Player is within their grace period
+            return true;
+        }
+
+        if (gracePeriodEnd > 0) {
```

Comment:
Im a little confused for this

We are firstly checking
if the graceperiod time end is higher or is equal to current time then we return yes it is
but then if its just higher we remove it?

Reply:
First check checks that the graceperiod is still active, if it is the code returns. If the code doesn't return then we check if the value is above zero, meaning there was an active graceperiod which has lapsed and can be removed from the map. This could be changed to give the boxed object and do a null check, though it's all the same net result?